src/data/feature_index.js

```diff
@@ -146,12 +146,12 @@ class FeatureIndex {
                 filter,
                 params.layers,
                 styleLayers,
-                (feature: VectorTileFeature, styleLayer: StyleLayer) => {
+                (feature: VectorTileFeature, styleLayer: StyleLayer, id: ?string | number) => {
                     if (!featureGeometry) {
                         featureGeometry = loadGeometry(feature);
                     }
                     let featureState = {};
-                    if (feature.id) {
+                    if (id) {
```

Comment:
Should this be `id !== undefined` to handle 0? (this would have been a pre-existing bug)

Reply:
Good catch, will fix!